### ISSUE TITLE
DROP TABLE IF EXISTS syntax only works on SQL Server >2016

### ISSUE THREAD

**Issue by joncooper** [mikaelene/dbt-sqlserver#20]
The syntax `DROP TABLE IF EXISTS ...` works only on SQL Server >2016

https://github.com/mikaelene/dbt-sqlserver/blob/fb228938805289b1ba84bf21534730d3e8d5a63c/dbt/include/sqlserver/macros/adapters.sql#L44

Thus dbt-sqlserver will currently fail on previous versions (mine is 2012) with something like:

```
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'if'. (156) (SQLExecDirectW)")
```

I'm happy to submit a PR, but is there an overall test suite that I can run somehow to verify other macros aren't similarly incompatible? I am not familiar enough with dbt at the moment to know how one tests an adapter.

**Comment by mikaelene** [comment]
Hi! There are a test-repo. Not sure how up to date it is https://github.com/fishtown-analytics/dbt-integration-tests/ . You are most welcome to submit a PR. I know there might be more stuff that does not work in earlier versions of sql server. I think some of them are solved in the other adapter. Dbt-mssql

**Comment by mikaelene** [comment]
One way to test if you don’t want to use the full integration test is to create a project with all kind of materialzations and run the dbt commands. Seeds, run, test etc. Not the most elegant solution but better then nothing

**Comment by swanderz** [comment]
@joncooper neither does it work for Synapse... here was my general workaround
```jinja
if object_id ('{{ relation.schema }}.{{ relation.identifier }}','U') is not null
    begin
    drop {{ relation.type }} {{ relation.schema }}.{{ relation.identifier }}
    end
```

**Comment by mikaelene** [comment]
You are all welcome to submit a pull request. Support for Synapse would be great 😀

**Comment by mikaelene** [comment]
This is solved by PR #57
